sdcm/utils/docker_utils.py

```diff
@@ -176,15 +176,23 @@ def _get_attr_for_name(instance: object,
 
         return default
 
-    @staticmethod
-    def get_container(instance: object, name, raise_not_found_exc: bool = True) -> Optional[Container]:
+    @classmethod
+    def get_container(cls, instance: object, name, raise_not_found_exc: bool = True) -> Optional[Container]:
         assert name is not None, "None is not allowed as a container name"
 
         container = instance._containers.get(name)
         if container is None and raise_not_found_exc:
             raise NotFound(f"There is no container `{name}' for {instance}.")
         if container:
-            container.reload()
+            try:
+                container.reload()
+            except OSError:
+                LOGGER.warning("Container failed to refresh, checking docker connection...")
+                docker_client = cls.get_docker_client(instance, name)
```

Comment:
`cls.get_docker_client` if it gets `name` param it runs `cls.get_container` - so we have recurrence here :)
I propose to not provide name for `cls.get_docker_client` in that case.

Reply:
I've added code for that, let's see how this will behave now.